### PR TITLE
Rem 267

### PR DESCRIPTION
**Pull request by Alinadimmokdad**
fi-dimensions

### CODE REVIEW

**Comment by NoraBitarSM** [comment]
functionality tested

File: src/pages/fi-dimensions/index.js [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+        filteredList.forEach(obj => {
+          if (obj.value && !isNaN(obj.value) && obj.value.trim() !== '') {
+            myObject[obj.key] = parseInt(obj.value)
+          } else {
+            myObject[obj.key] = obj.value
+          }
+        })
+
+        formik.setValues(myObject)
+        console.log('obj', myObject)
```
[the remark this answers is not in the document]
Comment (Alinadimmokdad) [reply src/pages/fi-dimensions/index.js]: removed

File: src/pages/fi-dimensions/index.js [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    tpaDimension12: null,
+    tpaDimension13: null,
+    tpaDimension14: null,
+    tpaDimension15: null,
+    tpaDimension16: null,
+    tpaDimension17: null,
+    tpaDimension18: null,
+    tpaDimension19: null,
+    tpaDimension20: null
+  })
```
Comment (RebeccaFrem02) [inline src/pages/fi-dimensions/index.js:44]: do we need this state?

File: src/pages/fi-dimensions/index.js [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+
+  const postDimensionSettings = obj => {
+    var dataToPost = [{ key: 'DimCount', value: obj.DimCount }]
+    for (let i = 1; i <= 20; i++) {
+      const dimKey = `tpaDimension${i}`
+      if (obj[dimKey] !== undefined) {
+        dataToPost.push({ key: dimKey, value: obj[dimKey] })
+      }
+    }
+    console.log(dataToPost)
```
Comment (RebeccaFrem02) [inline src/pages/fi-dimensions/index.js:114]: do we need this console?

File: src/pages/fi-dimensions/index.js [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+      extension: SystemRepository.Defaults.set,
+      record: JSON.stringify({ sysDefaults: dataToPost })
+    })
+      .then(res => {
+        toast.success('Record Successfully Updated')
+      })
+      .catch(error => {})
+  }
+
+  const handleSubmit = () => {
```
Comment (RebeccaFrem02) [inline src/pages/fi-dimensions/index.js:125]: is there a need to this function? can't we just call formik.handleSubmit() where we need it directly?

File: src/pages/fi-dimensions/index.js [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+  const handleDimCountBlur = () => {
+    if (tempDimCount !== null) {
+      formik.setFieldValue('DimCount', tempDimCount)
+    }
+  }
+
+  return (
+    <VertLayout>
+      <Grow>
+        <Grid container spacing={3} width={900} sx={{ marginLeft: '0.5rem' }}>
```
Comment (RebeccaFrem02) [inline src/pages/fi-dimensions/index.js:152]: do we need the width and margin here?

File: src/pages/fi-dimensions/index.js [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+    }
+  }, [formik.values.DimCount])
+
+  const handleDimCountChange = event => {
+    setTempDimCount(event.target.value)
+  }
+
+  const handleDimCountBlur = () => {
+    if (tempDimCount !== null) {
+      formik.setFieldValue('DimCount', tempDimCount)
```
Comment (hadichahine) [inline src/pages/fi-dimensions/index.js:145]: pass event.target.value directly without temp

File: src/pages/fi-dimensions/index.js [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+  return (
+    <VertLayout>
+      <Grow>
+        <Grid container spacing={3} width={900} sx={{ marginLeft: '0.5rem' }}>
+          <Grid item xs={12} sx={{ marginTop: '0.5rem' }}>
+            <CustomTextField
+              name='DimCount'
+              label={_labels.DimCount}
+              value={tempDimCount === null ? formik.values.DimCount : tempDimCount}
+              onChange={handleDimCountChange}
```
Comment (hadichahine) [inline src/pages/fi-dimensions/index.js:158]: Remove this

File: src/pages/fi-dimensions/index.js [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+        toast.success('Record Successfully Updated')
+      })
+      .catch(error => {})
+  }
+
+  const handleSubmit = () => {
+    formik.handleSubmit()
+  }
+
+  useEffect(() => {
```
Comment (hadichahine) [inline src/pages/fi-dimensions/index.js:129]: Move it into onBlur

File: src/pages/fi-dimensions/index.js [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+    const dimCount = formik.values.DimCount
```
Comment (hadichahine) [inline src/pages/fi-dimensions/index.js:130]: Dim count will be event.target.value

File: src/pages/fi-dimensions/index.js [hunk 10]
```diff
@@ -0,0 +1,225 @@
...
+    for (let i = 1; i <= 20; i++) {
+      const dimKey = `tpaDimension${i}`
+      if (i > dimCount) {
+        formik.setFieldValue(dimKey, null)
+      }
+    }
+  }, [formik.values.DimCount])
+
+  const handleDimCountChange = event => {
```
Comment (hadichahine) [inline src/pages/fi-dimensions/index.js:139]: Remove

File: src/pages/fi-dimensions/index.js [hunk 11]
```diff
@@ -0,0 +1,225 @@
...
+import FormShell from 'src/components/Shared/FormShell'
+import { useForm } from 'src/hooks/form'
+import { Fixed } from 'src/components/Shared/Layouts/Fixed'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+
+const FiDimensions = () => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+
+  const [tempDimCount, setTempDimCount] = useState(null)
```
Comment (hadichahine) [inline src/pages/fi-dimensions/index.js:20]: Remove

File: src/pages/fi-dimensions/index.js [hunk 12]
```diff
@@ -0,0 +1,225 @@
...
+      postDimensionSettings(values)
+    }
+  })
+
+  useEffect(() => {
+    getDataResult()
+  }, [])
+
+  const getDataResult = () => {
+    const myObject = {}
```
Comment (hadichahine) [inline src/pages/fi-dimensions/index.js:78]: name it as values

File: src/pages/fi-dimensions/index.js [hunk 13]
```diff
@@ -0,0 +1,225 @@
...
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+
+const FiDimensions = () => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+
+  const [tempDimCount, setTempDimCount] = useState(null)
+
+  const [initialValues, setInitialData] = useState({
+    DimCount: null,
```
Comment (hadichahine) [inline src/pages/fi-dimensions/index.js:23]: Remove

**Review by hadichahine — CHANGES_REQUESTED** [review]

**Review by hadichahine — APPROVED** [review]

**Pull request merged**